cmd/dashboard-token-proxy/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+type GithubResponse struct {
+	AccessToken string `json:"access_token"`
+	Scope       string `json:"scope"`
+	TokenType   string `json:"token_type"`
+}
+
+type DashboardResponse struct {
+	AccessToken string `json:"access_token"`
+	Success     bool   `json:"success"`
+}
+
+var (
+	clientID         string
+	clientSecret     string
+	ghURL            string
+	authorizationURL string
+)
+
+func main() {
+	clientID = os.Getenv("CLIENT_ID")
+	clientSecret = os.Getenv("CLIENT_SECRET")
+	ghURL = os.Getenv("GH_BASE_URL")
+
+	authorizationURL = fmt.Sprintf("https://%s/login/oauth/access_token", ghURL)
+
+	http.HandleFunc("/", HandleTokenRequest)
+	// Determine port for HTTP service.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+	// Start HTTP server.
+	log.Printf("Listening on port %s", port)
+	if err := http.ListenAndServe(":"+port, nil); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func HandleTokenRequest(w http.ResponseWriter, r *http.Request) {
+	if clientID == "" || clientSecret == "" {
+		log.Printf("Client data is missing")
+		http.Error(w, "Client data is missing", http.StatusBadRequest)
+		return
+	}
+
+	client := &http.Client{}
+
+	err := r.ParseForm()
+	if err != nil {
+		log.Printf("failed to parse form: %v", err)
+		http.Error(w, "failed to parse form", http.StatusBadRequest)
+		return
+	}
+
+	code := r.Form["oauthz_code"][0]
+
+	data := url.Values{}
+	data.Add("code", code)
+	data.Add("client_id", clientID)
+	data.Add("client_secret", clientSecret)
+
+	req, err := http.NewRequest("POST", authorizationURL, strings.NewReader(data.Encode()))
+	if err != nil {
+		log.Printf("coudn't create Github request: %v", err)
```

Comment:
fix typo, also please use more official language. "Could not" instead of "Couldn't"